src/app/main/component/events/components/create-edit-events/create-edit-events.component.ts

```diff
@@ -31,82 +35,131 @@ export class CreateEditEventsComponent implements OnInit {
+  ngOnInit(): void {
+    this.titleForm = new FormControl('', [Validators.required, Validators.minLength(1), Validators.maxLength(70)]);
+  }
+
+  constructor(private eventService: EventsService, public router: Router, private injector: Injector) {
     this.quillModules = quillConfig;
     Quill.register('modules/imageResize', ImageResize);
+    this.snackBar = injector.get(MatSnackBarComponent);
+  }
+
+  public checkTab(tag: TagObj): void {
+    tag.isActive = !tag.isActive;
+  }
+
+  public checkForm(form: DateFormObj, ind: number): void {
+    this.dates[ind].date = form.date;
+    this.dates[ind].startDate = form.startTime;
+    this.dates[ind].finishDate = form.endTime;
+    this.dates[ind].onlineLink = form.onlineLink;
+  }
+
+  public checkStatus(event: boolean, ind: number): void {
+    this.dates[ind].valid = event;
   }
 
   public escapeFromCreateEvent(): void {
-    this.router.navigate(['/events']).catch((err) => console.error(err));
+    this.router.navigate(['/events']);
   }
 
-  changeToOpen(): void {
+  public changeToOpen(): void {
     this.isOpen = true;
   }
-  changeToClose(): void {
+
+  public changeToClose(): void {
     this.isOpen = false;
   }
 
-  setDateCount(ev: MatSelectChange): void {
-    this.dates.length = +ev.value.split(' ')[0];
+  public setDateCount(event: MatSelectChange): void {
+    this.dates.length = +event.value.split(' ')[0];
 
     for (let i = 0; i < this.dates.length; i++) {
       this.dates[i] = {
-        date: '',
+        date: null,
         startDate: '',
         finishDate: '',
         coordinatesDto: {
           latitude: null,
           longitude: null
         },
-        onlineLink: ''
+        onlineLink: '',
+        valid: false,
+        check: false
       };
     }
   }
 
-  getImageTosend(imageArr: Array<File>): void {
+  public getImageTosend(imageArr: Array<File>): void {
     this.imgArray = [...imageArr];
   }
 
-  public getDate(event: string, ind: number): void {
-    this.dates[ind].date = event;
+  public changedEditor(event: EditorChangeContent | EditorChangeSelection): void {
+    if (event.event !== 'selection-change') {
+      this.editorHTML = event.html;
+      this.contentValid = !(event.text.length < 20 || event.text.length > 63206);
+    }
   }
 
-  public setStartTime(time: string, ind: number): void {
-    this.dates[ind].startDate = time;
-  }
-  public setEndTime(time: string, ind: number): void {
-    this.dates[ind].finishDate = time;
+  public setCoordsOnlOff(event: OfflineDto, ind: number): void {
+    this.dates[ind].coordinatesDto.latitude = event.latitude;
+    this.dates[ind].coordinatesDto.longitude = event.longitude;
   }
 
-  public changedEditor(event: EditorChangeContent | EditorChangeSelection): void {
-    if (event.event !== 'selection-change') {
-      this.editorHTML = event.html;
+  private checkDates() {
+    this.dates.forEach((item) => {
+      item.check = !item.valid;
+    });
+
+    for (const value of this.dates) {
```

Comment:
remove for..of cycle here, use some method for dates array
this.checkdates = !this.dates.some((item) => !item.valid)